### ISSUE TITLE
service resource not syncing when it has ipFamilies and related keys

### ISSUE THREAD

**Issue by jurgen-weber-deltatre** [argoproj/argo-cd#7609]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

![image](https://user-images.githubusercontent.com/77422409/140003869-e711780d-5f59-43cf-9387-67964d094750.png)

if a service resource explicitly has the following keys in it, it is perpetually always syncing|

```
ipFamilies:
  - IPv4
ipFamilyPolicy: SingleStack
```

**To Reproduce**

I upgraded from ingress nginx 3.39.0 to 4.0.6 and a service resource in there has the keys. But create any service resource with these keys

**Expected behavior**

I expect the app/resource to sync cleaning and go green.

**Screenshots**

above

**Version**

```shell
 $ argocd version
argocd-server: v2.1.6+a346cf9
  BuildDate: 2021-10-28T19:59:40Z
  GitCommit: a346cf933e10d872eae26bff8e58c5e7ac40db25
  GitTreeState: clean
  GoVersion: go1.16.5
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v4.2.0 2021-06-30T22:49:26Z
  Helm Version: v3.6.0+g7f2df64
  Kubectl Version: v0.21.0
  Jsonnet Version: v0.17.0

$ kubectl version
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.7-eks-d88609", GitCommit:"d886092805d5cc3a47ed5cf0c43de38ce442dfcb", GitTreeState:"clean", BuildDate:"2021-07-31T00:29:12Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jessesuen** [comment]
> Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.7-eks-d88609", GitCommit:"d886092805d5cc3a47ed5cf0c43de38ce442dfcb", GitTreeState:"clean", BuildDate:"2021-07-31T00:29:12Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}

The ipFamilies feature is beta in K8s v1.21, which means it won't be enabled by default in v1.20 without some k8s API server flags.

So I believe you need to upgrade your K8s cluster to get this feature.

**Comment by jurgen-weber-deltatre** [comment]
right, so it is a chart templating problem.

Thanks

**Comment by parberge** [comment]
@jurgen-weber-deltatre Care to share how you resolved it? We see the issue in Kubernetes 1.20 (but not 1.19).

**Comment by jurgen-weber-deltatre** [comment]
yeah, basically the chart is attempting to apply something the API doesn't support, so I resolved on the chart side with a kuberenetes version if statement.
